src/vs/workbench/services/themes/browser/workbenchThemeService.ts

```diff
@@ -72,6 +75,8 @@ export class WorkbenchThemeService implements IWorkbenchThemeService {
 
 	private colorThemeStore: ColorThemeStore;
 	private currentColorTheme: ColorThemeData;
+	private autoSwitchColorTheme = DEFAULT_THEME_AUTO_SWITCH_SETTING_VALUE;
+	private autoSwitchColorThemeListener: MediaQueryList | undefined = undefined;
```

Comment:
I think `IDisposable` does not work here because `matchMedia` returns a `MediaQueryList`. I will merge it into one but will leave the `MediaQueryList | undefined`